geonetwork/gn3-migration/schema-validation/fix-mcp-schema-validation.xsl

```diff
@@ -0,0 +1,201 @@
+          <xsl:apply-templates select="*|node()"/>
+        </mcp:MD_Commons>
+    </xsl:if>
+  </xsl:template>
+
+  <!-- Required attribute `<gml:id>` is generated -->
+  <xsl:template match="gml:Polygon[not(@gml:id)]">
+    <gml:Polygon gml:id="{generate-id(.)}">
+      <xsl:apply-templates select="@*|node()"/>
+    </gml:Polygon>
+  </xsl:template>
+
+  <!-- Empty `<gco:Real>` element removed  -->
+  <xsl:template match="gmd:EX_VerticalExtent/gmd:minimumValue">
+    <xsl:copy>
+      <xsl:if test="./gco:Real/*[not(*) and not(normalize-space())]">
+        <gmd:minimumValue gco:nilReason="unknown" />
+      </xsl:if>
+    </xsl:copy>
+  </xsl:template>
+
+  <!-- Empty `<gco:Real>` element removed  -->
+  <xsl:template match="gmd:EX_VerticalExtent/gmd:maximumValue">
+    <xsl:copy>
+      <xsl:if test="./gco:Real/*[not(*) and not(normalize-space())]">
+        <gmd:maximumValue gco:nilReason="unknown" />
+      </xsl:if>
+    </xsl:copy>
+  </xsl:template>
+
+  <!-- Empty `<gco:Real>` element removed  -->
+  <xsl:template match="gmd:MD_DigitalTransferOptions/gmd:transferSize">
+    <xsl:copy>
+      <xsl:if test="./gco:Real/*[not(*) and not(normalize-space())]">
+        <gmd:transferSize gco:nilReason="unknown" />
+      </xsl:if>
+    </xsl:copy>
+  </xsl:template>
+
+  <!-- Added required child element `<gml:end/>` -->
+  <xsl:template match="gmd:extent/gml:TimePeriod" xmlns:mcp="http://schemas.aodn.org.au/mcp-2.0">
+    <xsl:copy>
+      <xsl:choose>
+        <xsl:when test="(./gml:beginPosition) and (./gml:end) and not(./gml:endPosition)">
+          <xsl:apply-templates select="@* | gml:beginPosition" />
+          <gml:endPosition>
+            <xsl:apply-templates select="./gml:end/gml:TimeInstant/gml:timePosition/text()" />
+          </gml:endPosition>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:apply-templates select="@* | node()" />
+          <xsl:choose>
+            <xsl:when test="(./gml:begin) and not(./gml:end)">
+              <gml:end />
+            </xsl:when>
+            <xsl:when test="(./gml:beginPosition) and not(./gml:endPosition)">
+              <gml:endPosition />
+            </xsl:when>
+          </xsl:choose>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:copy>
+  </xsl:template>
+
+  <!-- Replaced `:` with `/` to make it valid URL -->
+  <xsl:template match="*[local-name() = 'vocabularyTermURL']/gmd:URL[.='5cc12a4f-6784-372a-801a-09d227bc67c4:concept:477']">
+    <gmd:URL>
+        <xsl:value-of select="translate(text(),':','/')" />
+    </gmd:URL>
+  </xsl:template>
+
+  <!-- Replace `<gmd:alternativeTitle>` with `<gmd:alternateTitle>` -->
+  <xsl:template match="gmd:CI_Citation/gmd:alternativeTitle">
+    <gmd:alternateTitle>
+      <xsl:apply-templates select="@*|node()"/>
+    </gmd:alternateTitle>
+  </xsl:template>
+
+<!-- Missing `<gmd:date>` element added  -->
+  <xsl:template match="gmd:CI_Date">
+    <xsl:copy>
+      <xsl:if test="not(./gmd:date)">
+        <gmd:date />
```

Comment:
As discussed, It would be best to close the `gmd:CI_Date` rather than its child to avoid issues in 19115-3.2018 after the transform.